analyzer/runtime/evm/erc165.go

```diff
@@ -0,0 +1,79 @@
+package evm
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/oasisprotocol/oasis-core/go/common/errors"
+
+	"github.com/oasisprotocol/oasis-indexer/analyzer/evmabi"
+	"github.com/oasisprotocol/oasis-indexer/log"
+	"github.com/oasisprotocol/oasis-indexer/storage/oasis/nodeapi"
+)
+
+func InterfaceID(abi *abi.ABI) []byte {
+	id := []byte{0, 0, 0, 0}
+	for _, m := range abi.Methods {
+		id[0] ^= m.ID[0]
+		id[1] ^= m.ID[1]
+		id[2] ^= m.ID[2]
+		id[3] ^= m.ID[3]
+	}
+	return id
+}
+
+var (
+	InvalidInterfaceID             = []byte{0xff, 0xff, 0xff, 0xff}
+	ERC165InterfaceID              = InterfaceID(evmabi.ERC165)
+	ERC721InterfaceID              = InterfaceID(evmabi.ERC721)
+	ERC721TokenReceiverInterfaceID = InterfaceID(evmabi.ERC721TokenReceiver)
+	ERC721MetadataInterfaceID      = InterfaceID(evmabi.ERC721Metadata)
+	ERC721EnumerableInterfaceID    = InterfaceID(evmabi.ERC721Enumerable)
```

Comment:
Consider providing the actual hex value in a comment, for ctrl+F debugging

Reply:
let's have that association in the tests so that they're programmatically verified

Reply:
won't provide. marking as resolved :stuck_out_tongue: